spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/observation/AppDeployerDocumentedObservation.java

```diff
@@ -0,0 +1,353 @@
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return Tags.values();
+		}
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	},
+
+	/**
+	 * Observation created upon asking for logs of deployed applications.
+	 */
+	DEPLOYER_SCALE_OBSERVATION {
+		@Override
+		public Class<? extends Observation.ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultAppDeployerObservationConvention.class;
+		}
+
+		@Override
+		public String getName() {
+			return "spring.cloud.deployer.scale";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "scale";
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return KeyName.merge(Tags.values(), ScaleTags.values());
+		}
+
+		@Override
+		public Event[] getEvents() {
+			return Events.values();
+		}
+	};
+
+	enum Tags implements KeyName {
+
+		/**
+		 * Name of the platform to which apps are being deployed.
+		 */
+		PLATFORM_NAME {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.name";
+			}
+		},
+
+		/**
+		 * ID of the deployed application.
+		 */
+		APP_ID {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.app.id";
+			}
+		},
+
+		/**
+		 * Name of the deployed application.
+		 */
+		APP_NAME {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.app.name";
+			}
+		},
+
+		/**
+		 * Group of the deployed application.
+		 */
+		APP_GROUP {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.app.group";
+			}
+		},
+
+		/**
+		 * CloudFoundry API URL.
+		 */
+		CF_URL {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.cf.url";
+			}
+		},
+
+		/**
+		 * CloudFoundry org.
+		 */
+		CF_ORG {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.cf.org";
+			}
+		},
+
+		/**
+		 * CloudFoundry space.
+		 */
+		CF_SPACE {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.cf.space";
+			}
+		},
+
+		/**
+		 * Kubernetes API URL.
+		 */
+		K8S_URL {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.k8s.url";
+			}
+		},
+
+		/**
+		 * Kubernetes namespace.
+		 */
+		K8S_NAMESPACE {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.platform.k8s.namespace";
+			}
+		},
+
+	}
+
+	enum ScaleTags implements KeyName {
+
+		/**
+		 * Scale command deployment id.
+		 */
+		DEPLOYER_SCALE_DEPLOYMENT_ID {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.scale.deploymentId";
+			}
+		},
+
+		/**
+		 * Scale count.
+		 */
+		DEPLOYER_SCALE_COUNT {
+			@Override
+			public String asString() {
+				return "spring.cloud.deployer.scale.count";
+			}
+		}
+
+	}
+
+	enum Events implements Event {
+
+		/**
+		 * When deployer started deploying the application.
```

Comment:
This is used also by undeploy. Should we have another one for undeploy or just update the docs here?

Reply:
I'm updating the docs